toolbox/src/DiscreteFilter.cpp

```diff
@@ -116,6 +118,11 @@ bool DiscreteFilter::initialize(BlockInformation* blockInfo, wbt::Error* error)
     // Create the filter object
     // ========================
 
+    // Get the width of the signals
+    unsigned inputSignalWidth  = blockInfo->getInputPortWidth(INPUT_IDX_SIGNAL);
+    unsigned outputSignalWidth = blockInfo->getOutputPortWidth(OUTPUT_IDX_SIGNAL);
+    assert(inputSignalWidth == outputSignalWidth);
```

Comment:
It crashes. In my opinion having asserts is good even in this case. When people work on controllers (downstream) the code should be always set in release. Debug should be enabled only when developing the block themselves. @francesco-romano